x-pack/elastic-agent/docs/elastic-agent.asciidoc

```diff
@@ -1,78 +1,255 @@
+
+E.g
+[source,yaml]
+-------------------------------------------------------------------------------------
+outputs:
+  default:
+    type: elasticsearch
+    hosts: [127.0.0.1:9200]
+    username: elastic
+    password: changeme
+
+  monitoring:
+    type: elasticsearch
+    api_key: VuaCfGcBCdbkQm-e5aOx:ui2lp2axTNmsyakw9tvNnw
+    hosts: ["localhost:9200"]
+    ca_sha256: "7lHLiyp4J8m9kw38SJ7SURJP4bXRZv/BNxyyXkCcE/M="
+-------------------------------------------------------------------------------------
+
+In the example above 2 outputs are provided. One called `default` and one called `monitoring`.
+Difference between them is in authentication method they are using, while first one uses username password pair second one contains api key.
+
+[NOTE]
+==============
+Configuration is invalid without definition of default output
+==============
+
+=== Enable/Disable Beats monitoring
+
+To disable or enable monitoring of managed Beats following section is available:
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+settings.monitoring:
+  # enabled turns on monitoring of running processes
+  enabled: true
+  # enables log monitoring
+  logs: true
+  # enables metrics monitoring
+  metrics: true
+  # specifies output to be used
+  use_output: monitoring
+-------------------------------------------------------------------------------------
+
+
+In case `settings.monitoring.enabled` is set to `false` monitoring of Beats is turned on and other options are ignored.
+If `settings.monitoring.enabled` is set to `true` Elastic Agent watches Metrics or Logs of Beats according to `settings.monitoring.metrics` or `settings.monitoring.logs` values.
+
+Having both set to `false` results in the same behavior as disabling monitoring.
+
+`use_output` option specifies output to which events will be send.
+
+=== Specifying data sources
+
+By default Elastic Agent ingests some system metrics. Its default configuration can look like this:
+
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+datasources:
+  - namespace: default
+    use_output: default
+    inputs:
+      - type: system/metrics
+        streams:
+          - metricset: cpu
+            dataset: system.cpu
+          - metricset: memory
+            dataset: system.memory
+          - metricset: network
+            dataset: system.network
+          - metricset: filesystem
+            dataset: system.filesystem
+-------------------------------------------------------------------------------------
+
+This configuration configures gathering cpu, memory, network and filesystem metrics and sending them to default output.
+If `use_output` option is not specified, output called `default` is used.
+
+For more examples please refer to link:elastic-agent_configuration_example.yml[Elastic Agent Configuration Example]
```

Comment:
```suggestion
For more examples, see link:elastic-agent_configuration_example.yml[Elastic Agent Configuration Example]
```

Reply:
again, I don't think this link will work, but I can help you get it working after I see where the file lives.